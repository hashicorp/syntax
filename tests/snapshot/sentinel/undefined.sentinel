# https://developer.hashicorp.com/sentinel/docs/language/undefined

foo() else 42
foo.bar else ""
config["bad-key"] else "default"

// In more complex scenarios

foo() else 42 == 12
a = config.value else "default"
