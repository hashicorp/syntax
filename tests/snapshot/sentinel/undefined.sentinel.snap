># https://developer.hashicorp.com/sentinel/docs/language/undefined
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.sentinel comment.line.number-sign.sentinel
>
>foo() else 42
#^^^^^^ source.sentinel
#      ^^^^ source.sentinel keyword.control.sentinel
#          ^ source.sentinel
#           ^^ source.sentinel constant.numeric.number.sentinel
>foo.bar else ""
#^^^^^^^^ source.sentinel
#        ^^^^ source.sentinel keyword.control.sentinel
#            ^ source.sentinel
#             ^ source.sentinel string.quoted.double.untitled
#              ^ source.sentinel string.quoted.double.untitled
>config["bad-key"] else "default"
#^^^^^^^ source.sentinel
#       ^ source.sentinel string.quoted.double.untitled
#        ^^^^^^^ source.sentinel string.quoted.double.untitled
#               ^ source.sentinel string.quoted.double.untitled
#                ^^ source.sentinel
#                  ^^^^ source.sentinel keyword.control.sentinel
#                      ^ source.sentinel
#                       ^ source.sentinel string.quoted.double.untitled
#                        ^^^^^^^ source.sentinel string.quoted.double.untitled
#                               ^ source.sentinel string.quoted.double.untitled
>
>// In more complex scenarios
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.sentinel comment.line.double-slash.sentinel
>
>foo() else 42 == 12
#^^^^^^ source.sentinel
#      ^^^^ source.sentinel keyword.control.sentinel
#          ^ source.sentinel
#           ^^ source.sentinel constant.numeric.number.sentinel
#             ^ source.sentinel
#              ^ source.sentinel keyword.operator.symbol.sentinel
#               ^ source.sentinel keyword.operator.symbol.sentinel
#                ^ source.sentinel
#                 ^^ source.sentinel constant.numeric.number.sentinel
>a = config.value else "default"
#^^ source.sentinel
#  ^ source.sentinel keyword.operator.symbol.sentinel
#   ^^^^^^^^^^^^^^ source.sentinel
#                 ^^^^ source.sentinel keyword.control.sentinel
#                     ^ source.sentinel
#                      ^ source.sentinel string.quoted.double.untitled
#                       ^^^^^^^ source.sentinel string.quoted.double.untitled
#                              ^ source.sentinel string.quoted.double.untitled
>